src/HydraAuction/OnChain.hs

```diff
@@ -118,10 +119,11 @@ feeEscrowAddress = FeeEscrowAddress . validatorAddress . feeEscrowValidator
 
 -- Deposit
 
+-- HACK: compilies incorrectly without HasCallStack
```

Comment:
This should not be the fix. This has most likely something do to with inlining, or specialisation, as adding this random constraint seems to fix the issue.